libs/server/Objects/Hash/HashObjectImpl.cs

```diff
@@ -393,7 +393,7 @@ private void HashIncrement(ref ObjectInput input, ref SpanByteAndMemory output)
                 }
                 else
                 {
-                    if (!NumUtils.TryParse(incrSlice.ReadOnlySpan, out float incr))
+                    if (!NumUtils.TryParse(incrSlice.ReadOnlySpan, out double incr))
```

Comment:
If you change the datatype from `float` to `double` you have to change the logic on how the double is converted into a string. Because `result.ToString(CultureInfo.InvariantCulture)` will give value in explorational notation eg: 0.00001 double value will be shown as `1E-5` but when we do `HGET` it should show the value as `0.00001` not `1E-5`. You have to make changes in the below 2 places for that.

https://github.com/microsoft/garnet/blob/b2448121ce2ddc3c0850cfd12dea25c42fb3e529/libs/server/Objects/Hash/HashObjectImpl.cs#L419

https://github.com/microsoft/garnet/blob/b2448121ce2ddc3c0850cfd12dea25c42fb3e529/libs/server/Objects/Hash/HashObjectImpl.cs#L426

Reply:
Thanks @Vijay-Nirmal! I'll have a look separately.

Reply:
@TalZaccai I will create a PR for this

Reply:
![image](https://github.com/user-attachments/assets/8dfe9c5e-436b-4864-96a2-44fe695ed79b)

Reply:
What's the issue with HGET? It should show the number in this format

Reply:
@TalZaccai What's happening in Garnet
![image](https://github.com/user-attachments/assets/940624e1-e2d9-456f-b2f7-28853169a82a)

Whats expected in Redis
![image](https://github.com/user-attachments/assets/5ea22e30-c782-4035-a038-2a3e244d3d2b)

Reply:
Ah ok so there's a culture difference, I see both as #2

Reply:
This shouldn't be a culture difference. Did you use exactly `0.0000001`? Or you can notice this in any number even non decimal like just type a very big number with more than 32 digits and check